packages/developer-portal/src/components/form-wizard/form-wizard.tsx

```diff
@@ -0,0 +1,59 @@
+import { Steps } from '@reapit/elements'
+import React, { useState } from 'react'
+import { FormWizardPropsInterface } from './interface'
+import { FieldValues, SubmitHandler } from 'react-hook-form'
+import { FormWizardStep } from './form-wizard-step'
+
+export const FormWizard = <TFieldValues extends FieldValues>({
+  steps,
+  onSubmit,
+  showStepNumbers = true,
+  isSubmitting = false,
+  submitButtonText = 'Submit',
+  ...rest
+}: FormWizardPropsInterface<TFieldValues>) => {
+  const [currentStep, setCurrentStep] = useState<number>(0)
+  const isSubmitStep = currentStep === Object.keys(steps).length - 1
+  const [formValues, setFormValues] = useState<TFieldValues>({} as TFieldValues)
+
+  const previousStep = () => {
+    setCurrentStep(currentStep - 1)
+  }
+
+  const nextOrSubmit: SubmitHandler<TFieldValues> = (values) => {
+    if (isSubmitStep) {
+      onSubmit(formValues)
+      return Promise.resolve(formValues)
```

Comment:
What happens if the backend does return an error at this point (I know we've tried to marry up the validation rules so that shouldn't happen, but need to ensure its handled properly)